### ISSUE TITLE
[DX] place all theme-related functions in modulename.theme.inc file

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#537]
...so that theme developers have an easier time finding theme functions and preprocess functions they are looking to override. It would be good to standardize on the location of these functions to improve the theme developer experience with Backdrop.

To make this change:

Look through core for modules that have their theme functions in the .module file, and move them into a theme.inc file instead.  You'll need to update the hook_theme() implementation (it should remain in the .module file).  For each theme function defined, include the following line.
```    'file' => 'modulename.theme.inc',```

Preprocess functions may be tricker to move, see the layout.module file for a good example to follow.

Follow-up to https://github.com/backdrop/backdrop-issues/issues/486

**Comment by BWPanda** [comment]
PR is my first attempt at addressing this. I know there are errors (lots of them), but they don't appear when you clear the caches...

**Comment by BWPanda** [comment]
The above comment refers to errors on the Backdrop site. When you clear all caches, everything looks fine. But when you then refresh the page again, a bunch of errors appear.

Not to mention all the Travis CI errors...

**Comment by quicksketch** [comment]
This is looking really good! I like the looks of moving more code out of .module files, that will make for a lighter bootstrap and lower memory footprint.

> When you clear all caches, everything looks fine. But when you then refresh the page again, a bunch of errors appear.

I think that's because of this problem: https://github.com/backdrop/backdrop/pull/636#discussion_r22447796

**Comment by BWPanda** [comment]
Thanks @quicksketch. I've pushed a new commit that does as you suggested and also moves a few node and system theme functions I somehow missed before.

No more errors! (in Backdrop anyway; we'll see what Travis thinks...)

**Comment by jenlampton** [comment]
Great start :) It looks like there are just a few fails left.  Maybe run these last few tests locally and see if you get any useful errors on the verbose output.

Also, don't worry about moving the theme function in dblog module since we are trying to kill that one over in https://github.com/backdrop/backdrop-issues/issues/539

**Comment by BWPanda** [comment]
It looks like the exceptions all have to do with Views. I undid the changes I made to Views module, and also to DBlog as requested. Don't have a local server to test on, so just have to make another commit here to test. Happy to rebase before merging.

**Comment by BWPanda** [comment]
> Maybe run these last few tests locally and see if you get any useful errors on the verbose output.

Just realised what you meant by this - use the Testing module in Backdrop UI :smile:
Will try to remember that in future.

**Comment by BWPanda** [comment]
Hmm, Travis' test results for 'Fixed image test theming' show 2 failures for the 'User administration' test. When I run that test locally, I get 29 failures (on a new install)... Any idea why?

**Comment by jenlampton** [comment]
User administration test has random failures. Ignore those :)  We'll get 'em eventually!

**Comment by jenlampton** [comment]
Actually @BWPanda if you are around today this is the only issue (other than https://github.com/backdrop/backdrop-issues/issues/486) that I think we should push to get done today before we release.  If you are around and can rebase / retest that would be fab.

**Comment by BWPanda** [comment]
@jenlampton Am available and willing :) Will do.

**Comment by BWPanda** [comment]
Have merged with latest 1.x branch and rebased to squash all commits (hopefully I didn't revert any previous commits...). We'll see what Travis thinks.

**Comment by quicksketch** [comment]
We can do these conversions at any time and it might be better to wait on them in a minor release. Because of the theme registry, we can move these functions/template around as needed since they're never loaded directly. However from a maintenance-perspective, it would be nice if they were in the same place throughout 1.x.

So basically, no pressure if this doesn't get completed. Thanks again @BWPanda!

**Comment by BWPanda** [comment]
There are only 4 failures now from form.test. Not sure why as they should be affected... Everything else working though.

**Comment by jenlampton** [comment]
Thanks @BWPanda !  I just squashed your commits together, resolved a few new merge conflicts, and am now testing the new PR :)

**Comment by quicksketch** [comment]
Though Jen got the tests passing, it conflicted once we rearranged everything else :P
Filed another new PR at https://github.com/backdrop/backdrop/pull/680.

**Comment by quicksketch** [comment]
Got that done. Phew! That's the last 1.0.0 issue.
